mico-core/src/main/java/io/github/ust/mico/core/model/MicoService.java

```diff
@@ -67,51 +66,54 @@
      * A brief name for the service.
      */
     @ApiModelProperty(required = true)
+    @NotEmpty
     private String shortName;
 
-    /**
-     * The name of the artifact. Intended for humans.
-     */
-    @ApiModelProperty(required = true)
-    private String name;
-
     /**
      * The version of this service. Refers to GitHub release tag.
      */
     @ApiModelProperty(required = true)
+    @NotEmpty
     private String version;
 
     /**
-     * Human readable description of this service.
+     * Indicates where this service originates from, e.g.,
+     * GitHub (downloaded and built by MICO) or DockerHub
+     * (ready-to-use image).
      */
     @ApiModelProperty(required = true)
-    private String description;
+    @NotNull
+    private MicoServiceCrawlingOrigin serviceCrawlingOrigin = MicoServiceCrawlingOrigin.NOT_DEFINED;
+
+    // ----------------------
+    // -> Optional fields ---
+    // ----------------------
 
     /**
      * The list of interfaces this service provides.
+     * It can be omitted or be set to an empty list, but {@code null} is not allowed.
      */
-    @ApiModelProperty(required = true)
     @Relationship(type = "PROVIDES_INTERFACES", direction = Relationship.UNDIRECTED)
+    @NotNull
     private List<MicoServiceInterface> serviceInterfaces = new ArrayList<>();
 
     /**
-     * Indicates where this service originates from, e.g.,
-     * GitHub (downloaded and built by MICO) or DockerHub
-     * (ready-to-use image).
+     * The name of the artifact. Intended for humans.
      */
-    private MicoServiceCrawlingOrigin serviceCrawlingOrigin;
-
-
-    // ----------------------
-    // -> Optional fields ---
-    // ----------------------
+    private String name;
+    /**
+     * Human readable description of this service.
+     */
+    private String description;
```

Comment:
add to required properties

Reply:
Why should the *description* me mandatory?

Reply:
the UI wants to have a human readable description for every service